packages/ui/src/components/va-alert/VaAlert.vue

```diff
@@ -68,127 +66,110 @@
-  }
-
-  get alertStyle () {
-    let background = this.colorComputed
-    let boxShadow = 'none'
-    if (this.outline) {
-      background = 'transparent'
-    }
-    if (this.border) {
-      background = '#ffffff'
-      boxShadow = 'var(--va-alert-box-shadow)'
-    }
-    return {
-      border: this.outline && `1px solid ${this.colorComputed}`,
-      background: background,
-      boxShadow: boxShadow,
-      padding: this.dense && '0.25rem 0.75rem',
-    }
-  }
-
-  get contentStyle () {
-    let color = getTextColor(this.colorComputed)
-    if (this.outline) {
-      color = this.colorComputed
-    }
-    if (this.border) {
-      color = dark
-    }
-    return {
-      alignItems: this.center && 'center',
-      color: color,
-    }
-  }
-
-  get titleStyle () {
-    let color = getTextColor(this.colorComputed)
-    if (this.outline) {
-      color = this.colorComputed
-    }
-    if (this.border) {
-      color = this.colorComputed
-    }
-    return {
-      color: color,
-    }
-  }
+  props: {
+    color: ({ type: String as PropType<string>, default: 'primary' }),
+    title: ({ type: String as PropType<string>, default: '' }),
+    description: ({ type: String as PropType<string>, default: '' }),
+    icon: ({ type: String as PropType<string>, default: '' }),
+    closeText: ({ type: String as PropType<string>, default: '' }),
+    closeable: ({ type: Boolean as PropType<boolean>, default: false }),
+    dense: ({ type: Boolean as PropType<boolean>, default: false }),
+    outline: ({ type: Boolean as PropType<boolean>, default: false }),
+    center: ({ type: Boolean as PropType<boolean>, default: false }),
+    borderColor: ({ type: String as PropType<string>, default: '' }),
+    border: ({
+      type: String as PropType<string>,
+      default: '',
+      validator: (value: string) => ['top', 'right', 'bottom', 'left', ''].includes(value),
+    }),
+    modelValue: ({
+      type: Boolean as PropType<boolean>,
+      default: true,
+    }),
+  },
+  setup (props, { slots, emit }) {
+    const dark = 'var(--va-dark)'
+
+    const { getColor } = useColors()
+    const colorComputed = computed(() => getColor(props.color))
+
+    const hasIcon = computed(() => props.icon || slots.icon)
+    const hasTitle = computed(() => props.title || slots.title)
+    const borderClass = computed(() => `va-alert__border--${props.border}`)
+    const closeIcon = computed(() => props.closeText || 'close')
+    const alertStyle = computed(() => {
+      let background = colorComputed.value
+      let boxShadow = 'none'
+      if (props.outline) {
+        background = 'transparent'
+      }
+      if (props.border) {
+        background = '#ffffff'
+        boxShadow = 'var(--va-alert-box-shadow)'
+      }
+      return {
+        border: props.outline && `1px solid ${colorComputed.value}`,
+        padding: props.dense && '0.25rem 0.75rem',
+        background,
+        boxShadow,
+      }
+    })
+    const contentStyle = computed(() => {
```

Comment:
It would be cool to have styling logic hidden in some useAlertStyle hook… It's not a requirement, but hiding this logic will make this component even more simpler.

Reply:
Done